internal/app/query/query_service.go

```diff
@@ -135,7 +135,48 @@ func (s *queryService) QueryTicketIds(req *pb.QueryTicketIdsRequest, responseSer
 }
 
 func (s *queryService) QueryBackfills(req *pb.QueryBackfillsRequest, responseServer pb.QueryService_QueryBackfillsServer) error {
-	return status.Error(codes.Unimplemented, "not implemented")
+	ctx := responseServer.Context()
+	pool := req.GetPool()
+	if pool == nil {
+		return status.Error(codes.InvalidArgument, ".pool is required")
+	}
+
+	pf, err := filter.NewPoolFilter(pool)
+	if err != nil {
+		return err
+	}
+
+	var results []*pb.Backfill
+	err = s.bc.request(ctx, func(backfills map[string]*pb.Backfill) {
+		for _, backfill := range backfills {
+			if pf.In(backfill) {
+				results = append(results, backfill)
+			}
+		}
+	})
+	if err != nil {
+		err = errors.Wrap(err, "QueryBackfills: failed to run request")
+		return err
+	}
+
+	stats.Record(ctx, backfillsPerQuery.M(int64(len(results))))
```

Comment:
I think it's worth putting _len(results)_ to a separate variable. You call it 4 times in this method.

Reply:
Fixed